theseus/src/launcher/download.rs

```diff
@@ -140,9 +141,12 @@ pub async fn download_assets(
     index: &AssetsIndex,
 ) -> crate::Result<()> {
     log::debug!("Loading assets");
```

Comment:
as mentioned, we should have a better way of updating the progress of all the profile-install related stuff. I do thing the approach used in this PR may not be the best way to do the loading bars (wrapping the concurrent futures stuff). see disc discussion